pontoon/api/views.py

```diff
@@ -0,0 +1,102 @@
+from datetime import datetime, timedelta
+
+from django.contrib.auth.decorators import login_required
+from django.http import JsonResponse
+from django.utils.timezone import make_aware
+from django.views.decorators.http import require_GET
+
+from pontoon.actionlog.models import ActionLog
+from pontoon.base.models import Project
+
+
+@require_GET
+@login_required(redirect_field_name="", login_url="/403")
+def get_user_actions(request, date, slug=None):
+    try:
+        start_date = make_aware(datetime.strptime(date, "%Y-%m-%d"))
+    except ValueError:
+        return JsonResponse(
+            {
+                "error": "Invalid date format. Please use YYYY-MM-DD.",
+            },
+            status=400,
+        )
+
+    end_date = start_date + timedelta(days=1)
+
+    actions = ActionLog.objects.filter(
+        created_at__gte=start_date,
+        created_at__lt=end_date,
+        action_type__startswith="translation:",
+    )
+
+    if slug:
+        try:
+            project = Project.objects.get(slug=slug)
+        except Project.DoesNotExist:
+            return JsonResponse(
+                {
+                    "error": "Project not found. Please use a valid project slug.",
+                },
+                status=404,
+            )
+        actions = actions.filter(translation__entity__resource__project=project)
+
+    actions = actions.prefetch_related(
+        "performed_by__profile",
+        "translation__entity__resource__project",
+        "translation__errors",
+        "translation__warnings",
+        "translation__locale",
+        "entity__resource__project",
+        "locale",
+    )
+
+    output = []
+
+    for action in actions:
+        user = action.performed_by
+        locale = action.locale or action.translation.locale
+        entity = action.entity or action.translation.entity
+        resource = entity.resource
+        project = resource.project
+
+        data = {
+            "type": action.action_type,
+            "date": action.created_at,
+            "user": {
+                "pk": user.pk,
+                "name": user.display_name,
+                "system_user": user.profile.system_user,
+            },
+            "locale": {
+                "pk": locale.pk,
+                "code": locale.code,
+                "name": locale.name,
+            },
+            "entity": {
+                "pk": entity.pk,
+                "key": entity.key,
+            },
+            "resource": {
+                "pk": resource.pk,
+                "path": resource.path,
+                "format": resource.format,
+            },
+            "project": {
+                "pk": project.pk,
+                "slug": project.slug,
+                "name": project.name,
+            },
```

Comment:
This seems like a lot of duplication, but presumably the data volumes here aren't large enough to really care?